### ISSUE TITLE
Unable to associate/delete emoji id record to a yat

### ISSUE THREAD

**Issue by z0rg0n** [z0rg0n/multiyat#3]
Searching the yat info of 😶💵😶 returns:
`{'error': None,
 'result': [{'data': '888tNkZrPN6JsEgekjMnABU4TBzc2Dt29EPAvkRxbANsAnjyPbb3iQ1YBRk1UXcdRsiKc9dhwMVgN5S9cQUiyoogDavup3H',
             'hash': '64afb085f00d56d4a0946e5dd37da4a157c1ef31231db0768e555d8db266df65',
             'tag': '0x1001'},
            {'data': 'https://getmonero.org',
             'hash': '6c1be3c4f1c9046400fbc219afb3d28f493a1134a01a2acb6f46b969758fdfb7',
             'tag': '0x0004'}],
 'stats': [{'description': 'Number of times emoji was looked up via API',
            'finish_date': '2022-02-27T22:26:55.661718680Z',
            'key': '😶💵😶',
            'metric': 'api_emoji_lookups',
            'start_date': '2022-01-30T22:26:55.661718050Z',
            'value': 653}],
 'status': True}
`

Docs [here](https://api-docs.y.at/docs/integration_general#adding-an-emoji-id-record-to-the-yat) outline that data can be stored to a yat.
[Tags are used to indicate what kind of data is being stored.](https://api-docs.y.at/docs/categories)
  
From the info in the 😶💵😶 yat I have an example of a Monero address 'tag' which is 0x1001.
From that info and the doc above I made emoji_id.store() in emoji_id.py.
Running this however returns "<Response [405]>" so something isn't right.

**Comment by z0rg0n** [comment]
The "Store value under EmojiId key" section of the API guides show I need to POST /emoji_id/{eid}/json/{key}

emoji_id is easy enough, I just pass a yat. But what is this key? Is it the "Enumerated Values" also listed in that seciton?
I tried to send them as they key and they gave the same error, <Response [405]>.

Since that error really isn't giving me anything useful I'm thinking I'm just way off on how I'm posting this data.

**Comment by z0rg0n** [comment]
Found these sections of the API docs:
https://api-docs.y.at/docs/integration_general#adding-an-emoji-id-record-to-the-yat
https://api-docs.y.at/docs/api-ref#edit-emojiid

Seems I was way off and an edit should be straight forward enough.

**Comment by z0rg0n** [comment]
Updated with fixed headers from https://github.com/z0rg0n/multiyat/issues/1 but still returning a 405 error.

URL I'm sending the request to:
`https://a.y.at/emoji_id/🤘🐺🤘`
sent dictionary data (converted to json with requests json=):
`{'insert': {'data': '48bPRVkgvHwjG2VUTkPLaGazynZ6RxETuNGsYZNBrtb7ZkAUqY1NE2iGqoLd8EFsvhbDGW8gNb96Jce8fg2aiY8A5mbd8zf', 'tag': '0x1001'}}`
sent header:
`{'Content-Type': 'application/json', 'X-Api-Key': '245_########################################################}`

The delete function has the same issue.


405 error seems to be that it sees my request but it isn't an allowed request.

My best guess is the cause has something to do with the headers I'm sending.
Other guesses include this operation is blocked.
Or maybe it has something to do with how the json module converts dictionary to json strings.

However I've kind of ruled out the last option since I tried to send this in my post:
`string_del = ('"delete":["cc99003dc1d247ee990267f5e7c0049971c698b1daefc31fbffa3c2f674c8a32"]')`
Which still gave me the same error.

**Comment by Krakaw** [comment]
The data needs to be in an array
```json
{'insert': [{'data': '48bPRVkgvHwjG2VUTkPLaGazynZ6RxETuNGsYZNBrtb7ZkAUqY1NE2iGqoLd8EFsvhbDGW8gNb96Jce8fg2aiY8A5mbd8zf', 'tag': '0x1001'}]}
```

**Comment by z0rg0n** [comment]
I did have that wrong but looks like something else is going wrong as well.

To both the set function I've sent:
`{'insert': [{'data': '48bPRVkgvHwjG2VUTkPLaGazynZ6RxETuNGsYZNBrtb7ZkAUqY1NE2iGqoLd8EFsvhbDGW8gNb96Jce8fg2aiY8A5mbd8zf', 'tag': '0x1001'}]}`

Both as a string and a dictionary to requests.post( URL, **json:** and requests.post( URL, **data:** 
(Reflected currently in code)

I think the problem may be how the requests library handles nested json since I found this:
https://stackoverflow.com/questions/43600125/nested-dictionaries-to-json-for-the-post-request-python
However, the solution in the top post didn't work (also in current code). Will have to do some more research into nested json or start learning a language other than python.
